app/src/main/java/org/fossasia/openevent/app/event/detail/EventDetailPresenter.java

```diff
@@ -77,145 +40,71 @@ public void detach() {
     }
 
     @Override
-    public void loadEvent(long eventId, boolean forceReload) {
-        if(eventDetailView == null)
+    public void loadDetails(boolean forceReload) {
+        if (eventDetailView == null)
             return;
 
+        eventDetailView.showProgressBar(true);
+
         eventRepository
             .getEvent(eventId, forceReload)
+            .doOnComplete(() ->
+                loadAttendees(eventId, forceReload))
             .subscribe(this::processEventAndDisplay,
                 throwable -> {
                     if(eventDetailView == null)
                         return;
-                    eventDetailView.showEventLoadError(throwable.getMessage());
-                    hideProgressbar();
-                });
+                    eventDetailView.showError(throwable.getMessage());
+                    hideProgress(forceReload);
+            });
     }
 
     private void processEventAndDisplay(Event event) {
         if(eventDetailView == null)
             return;
 
         this.event = event;
-
-        showEventInfo(event);
-
-        List<Ticket> tickets = event.getTickets();
-
-        long totalTickets = 0;
-        if(tickets != null) {
-            for (Ticket thisTicket : tickets)
-                totalTickets += thisTicket.getQuantity();
-        }
-
-        TicketAnalyser.analyseTotalTickets(event);
-        event.totalTickets.set(totalTickets);
-        event.totalAttendees.set(totalAttendees);
-        event.checkedIn.set(checkedInAttendees);
-        event.totalSale.set(totalSales);
-
-        hideProgressbar();
-    }
-
-    private void showEventInfo(Event event) {
-        if(eventDetailView == null)
-            return;
-
         eventDetailView.showEvent(event);
+        // TODO: Add views for date of event and format it here
 
-        String[] startDate = event.getStartTime().split("T");
-        String[] endDate = event.getEndTime().split("T");
-
-        event.startDate.set(startDate[0]);
-        event.endDate.set(endDate[0]);
-        event.eventStartTime.set(endDate[1]);
+        ticketAnalyser.analyseTotalTickets(event);
     }
 
-    @Override
-    public void loadAttendees(long eventId, boolean forceReload) {
+    private void loadAttendees(long eventId, boolean forceReload) {
         if(eventDetailView == null)
             return;
 
         eventRepository
             .getAttendees(eventId, forceReload)
             .toList()
-            .subscribe(this::processAttendeesAndDisplay,
+            .subscribe(
+                attendees -> {
+                    if(eventDetailView == null)
+                        return;
+                    ticketAnalyser.analyseSoldTickets(event, attendees);
+                    hideProgress(forceReload);
```

Comment:
That's not a problem because the data loading part is only time consuming and we display other information about event asap along with the total details, sold details may follow some milliseconds later